### ISSUE TITLE
experimental::lexical_topic: my $_ in code-match

### ISSUE THREAD

**Issue by atoomic** [rurban/perl-compiler#299]
Env perl 5.22.0

Strangely this test is marked as todo in the coretesuite ( since test was added in 59f00321bbc2d046 ),
it's failing with perl and passing with perlcc.

At least this is difference of behavior between an uncompiled and compiled code

````
> perl -e 'no warnings qw{experimental::lexical_topic}; my $s = "ok\n"; my $_ = "not ok\n"; my $r = $s =~ /ok(?{ print qq[$_] })/;'
not ok
> perlcc -r -e 'no warnings qw{experimental::lexical_topic}; my $s = "ok\n"; my $_ = "not ok\n"; my $r = $s =~ /ok(?{ print qq[$_] })/;'
ok
````

**Comment by atoomic** [comment]
testc.sh 299

**Comment by rurban** [comment]
Yes, this is an old upstream bug fixed in cperl with https://github.com/perl11/cperl/commit/597e929c930304418f422b0c01683da6f285587c

**Comment by rurban** [comment]
Fixed with 1.52-22 (HINTS_HASH)
Note that the test will fail will blead. lexical_topic was removed.
